src/app/staticanalysis/modules/checksEffectsInteraction.js

```diff
@@ -0,0 +1,83 @@
+var name = 'Checks-Effects-Interaction pattern'
+var desc = 'Avoid potential reentrancy bugs'
+var categories = require('./categories')
+var common = require('./staticAnalysisCommon')
+var callGraph = require('./functionCallGraph')
+var AbstractAst = require('./abstractAstView')
+
+function checksEffectsInteraction () {
+  this.contracts = []
+
+  checksEffectsInteraction.prototype.visit = new AbstractAst().builder(
```

Comment:
Now I'm wondering which object is `checkEffectsInteraction.prototye`.  I'm also wondering if this assignment can create a new property in that object.